vortex-jni/src/dtype.rs

```diff
@@ -228,55 +228,45 @@ pub extern "system" fn Java_dev_vortex_jni_NativeDTypeMethods_getTimeUnit(
 ) -> jbyte {
     let dtype = unsafe { &*(dtype_ptr as *const DType) };
 
-    let DType::Extension(ext_dtype) = dtype else {
-        vortex_err!("DType should be an EXTENSION, was {dtype}").throw_illegal_argument(&mut env);
-        return -1;
-    };
+    try_or_throw(&mut env, |_| {
+        let DType::Extension(ext_dtype) = dtype else {
+            throw_runtime!("DType should be an EXTENSION, was {dtype}");
+        };
 
-    match TemporalMetadata::try_from(ext_dtype) {
-        Ok(temporal) => match temporal.time_unit() {
+        let temporal = TemporalMetadata::try_from(ext_dtype)?;
+        Ok(match temporal.time_unit() {
             TimeUnit::Ns => 0,
             TimeUnit::Us => 1,
             TimeUnit::Ms => 2,
             TimeUnit::S => 3,
             TimeUnit::D => 4,
-        },
-        Err(err) => {
-            err.throw_illegal_argument(&mut env);
-            -1
-        }
-    }
+        })
+    })
 }
 
 #[unsafe(no_mangle)]
 pub extern "system" fn Java_dev_vortex_jni_NativeDTypeMethods_getTimeZone<'local>(
     mut env: JNIEnv<'local>,
     _class: JClass<'local>,
     dtype_ptr: jlong,
-) -> JString<'local> {
+) -> jstring {
```

Comment:
`JString<'a>` is just some sugar on top of the raw type, there's no meaningful difference returning one or the other. The reason to return `jstring` is b/c it's really just an alias to a pointer and has `Default` implemented